s3compatapi/src/main/java/com/snowflake/s3compatapitestsuite/compatapi/S3CompatStorageClient.java

```diff
@@ -65,10 +65,10 @@ private AmazonS3 createS3Client(
         clientCfg.withSocketTimeout(TIME_OUT);
         clientCfg.withTcpKeepAlive(true);
         AmazonS3 s3Client = new AmazonS3Client(awsCredentialsProvide, clientCfg);
-        s3Client.setEndpoint(endpoint);
         if (region != null) {
             s3Client.setRegion(Region.getRegion(Regions.fromName(region)));
         }
+        s3Client.setEndpoint(endpoint);
```

Comment:
nit: Add a comment that order matters between setting region and endpoint and why here?